packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -347,14 +346,19 @@ export class ModelAuthTransformer extends Transformer {
     const rules = this.getAuthRulesFromDirective(directive);
     // Assign default providers to rules where no provider was explicitly defined
     this.ensureDefaultAuthProviderAssigned(rules);
+
+    // If owner auth for read, update, or delete is specified, owner auth for create must also be included
```

Comment:
This is a change in behavior, hence a breaking change. Automatically altering customer defined auth rules by adding new ones is something we should discuss. Imagine the following scenario: When you register into a school only the Administrative persons can create the record for you and you'll be the owner and as owner you'll be able to update your data, but you cannot enroll yourself. By implicitly adding new rules break this use case.

Reply:
This change doesn't break that use case. An administrator could still create a record and set the owner to a different user at which point that other user would be the owner. This change is in the case that you have an auth rule where you specify owner auth on an operation besides create. In that case, we don't store owner at all on the create and then when the protected operation is attempted, the conditional check fails because the record has no owner.

Reply:
But adding Create permission for the owner, isn't enabling the user to create their own record?

Reply:
Oh I see. I guess the bottom line is, if any sort of owner permissions are specified for the model, we need to make sure that an owner gets stored when the model is created. Currently the Todo example [here](https://aws-amplify.github.io/docs/cli-toolchain/graphql#owner-authorization) doesn't work because read is owner protected but no owner is stored in the record when it's created.

Maybe with group auth there's no way to perform this check automatically? At the very least, we can't have a broken example (and a rather basic one at that) in our docs

Reply:
Yes I agree, we have to fix the example in the docs.

Reply:
Talked to @SwaySway about it and we agreed that the logic of the current implementation is flawed but the long-term fix is the auth refactor that he's working on. So I'll open a PR to fix the docs and we can discard this PR

Reply:
See: https://github.com/aws-amplify/docs/pull/1170